src/intersections/triangles.jl

```diff
@@ -122,9 +122,70 @@ function intersecttype(f::Function, s::Segment{3,T}, t::Triangle{3,T}) where {T}
       # rejection 1, coplanar segment
       return NoIntersection() |> f
     end
-  end      
+  end
 
   λ = clamp(wᵥ / (wᵥ - sᵥ), zero(T), one(T))
 
   return IntersectingSegmentTriangle(s(λ)) |> f
-end
\ No newline at end of file
+end
+
+"""
+    intersecttype(s, t)
+
+Calculate the intersection of a ray and triangle in 3D
+and apply function `f` to it.
+
+## References
+
+* Möller, T. and Trumbore, B., 1997. [Fast, minimum storage ray-triangle
+  intersection. Journal of graphics tools]
+  (https://www.tandfonline.com/doi/abs/10.1080/10867651.1997.10487468)
+"""
+function intersecttype(f::Function, r::Ray{3,T}, t::Triangle{3,T}) where {T}
+  vₜ = vertices(t)
+  origin = r.p
+  D = r.v
```

Comment:
Let's try to call these vectors `o` and `d` in lowercase. We try to reserve uppercase for points.